lib/Twig/Environment.php

```diff
@@ -117,6 +119,40 @@ public function __construct(Twig_LoaderInterface $loader = null, $options = arra
     }
 
     /**
+     * Returns deprecations for templates contained in a directory.
+     *
+     * @param string $dir The directory where templates are stored
+     * @param string $ext Limit the loaded templates by extension
+     *
+     * @return An array of deprecations
+     */
+    public function getDeprecations($dir, $ext = '.twig')
```

Comment:
I don't think this really belongs to the Twig environment. It should be built on top of it IMO (as done for the twig linting command of Symfony).
I suggest putting this in an dedicated utility class wrapping the environment